src/OpenLoco/src/GameCommands/Terraform/LowerRaiseLandMountain.cpp

```diff
@@ -0,0 +1,405 @@
+
+        auto height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::north);
+        adjustSurfaceSlopeWest(targetPos, height, removedBuildings);
+
+        *mtnToolHeightDiff -= kSmallZStep;
+
+        if (targetPos.y <= args.pointB.y)
+        {
+            *mtnToolHeightDiff += kSmallZStep;
+            if (targetPos.y < args.pointA.y)
+            {
+                *mtnToolHeightDiff += kSmallZStep;
+            }
+        }
+
+        height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::east);
+        adjustSurfaceSlopeSouth(targetPos, height, removedBuildings);
+    }
+
+    static void smoothenSurfaceWest(const Pos2& refPos, const Pos2& targetPos, const LowerRaiseLandMountainArgs& args, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        auto tile = TileManager::get(refPos);
+        auto surface = tile.surface();
+
+        auto height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::east);
+        adjustSurfaceSlopeNorth(targetPos, height, removedBuildings);
+
+        *mtnToolHeightDiff -= kSmallZStep;
+
+        if (targetPos.x <= args.pointB.x)
+        {
+            *mtnToolHeightDiff += kSmallZStep;
+            if (targetPos.x < args.pointA.x)
+            {
+                *mtnToolHeightDiff += kSmallZStep;
+            }
+        }
+
+        height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::south);
+        adjustSurfaceSlopeWest(targetPos, height, removedBuildings);
+    }
+
+    // 0x00462DCE
+    static uint32_t lowerRaiseLandMountain(const LowerRaiseLandMountainArgs& args, const uint8_t flags)
+    {
+        mtnToolXParams = (args.pointB.x << 16) | args.pointA.x;
+        mtnToolYParams = (args.pointB.y << 16) | args.pointA.y;
+        mtnToolGCFlags = flags;
+
+        if (flags & Flags::apply)
+        {
+            S5::getOptions().madeAnyChanges = 1;
+        }
+
+        mtnToolCost = 0;
+
+        // We keep track of removed buildings for each tile visited
+        // this prevents accidentally double counting their removal
+        // cost if they span across multiple tiles.
+        std::set<World::Pos3, LessThanPos3> removedBuildings{};
+
+        // Play sound if this is a company action
+        if ((flags & Flags::apply) && getCommandNestLevel() == 1 && getUpdatingCompanyId() != CompanyId::neutral)
+        {
+            const auto height = TileManager::getHeight(args.centre).landHeight;
+            Audio::playSound(Audio::SoundId::construct, World::Pos3(args.centre.x, args.centre.y, height));
+        }
+
+        // First, raise/lower the mountain's centre tile
+        {
+            auto result = adjustMountainCentre(args, removedBuildings, flags);
+            if (result != FAILURE)
+            {
+                mtnToolCost = *mtnToolCost + result;
+            }
+        }
+
+        if (!validCoords(args.pointA) || !validCoords(args.pointB))
```

Comment:
should really validate this before moving the center

Reply:
Not sure what you mean. This is intended to cover the use of the tool around map edges.

Reply:
Move it to before L252